go/arrow/array/struct.go

```diff
@@ -105,6 +107,36 @@ func (a *Struct) setData(data *Data) {
 	}
 }
 
+func (a *Struct) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+
+	tmp := make(map[string]interface{})
+	fieldList := a.data.dtype.(*arrow.StructType).Fields()
+	for j, d := range a.fields {
+		tmp[fieldList[j].Name] = d.getOneForMarshal(i)
```

Comment:
I don't think the arrow spec says struct names need to be unique, it might be worth checking for that case here and throwing an error OR using a list of tuple representation in that case.

Reply:
currently the Go implementation does not support duplicate field names in structs, only in schemas. So at least for now i'm guaranteed that there will be no duplicate field names until i make a change to support them in the struct type in general.

Reply:
hmm, OK, I might be wrong about the struct then, I might have been thinking about schema.